### PR TITLE
[Keyboard] Add Nozbe Macropad

### PR DESCRIPTION
**Pull request by rozPierog**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by drashna** [comment]
Please review the PR Checklist docs, and make the changes outlined there:
https://docs.qmk.fm/#/pr_checklist

Eg, you haven't followed it: 
* You're missing copyright headers for your c and h files
* You're using old style include guards (ifndef/define, instead of `#pragma once`)
* including default IS_COMMAND
* prevent stuck modifiers has not been a setting for 3+ years, now
* question marks in rules.mk file

These need to be resolved before the PR can be considered for merging

**Comment by rozPierog** [comment]
Thanks for the reply @drashna I've updated the files to be compliant with PR_checklist. I hope that I didn't miss anything

File: keyboards/handwired/nozbe_macro/config.h [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+#define DEBOUNCE 5
+
+/* Mechanical locking support. Use KC_LCAP, KC_LNUM or KC_LSCR instead in keymap */
+#define LOCKING_SUPPORT_ENABLE
+
+/* Locking resynchronize hack */
+#define LOCKING_RESYNC_ENABLE
+
+#ifdef RGB_DI_PIN
+#    define RGBLIGHT_ANIMATIONS
```
Comment (drashna) [inline keyboards/handwired/nozbe_macro/config.h:59]: Could you explicitly define which animations are enabled? 
```suggestion
#define RGBLIGHT_EFFECT_BREATHING
#define RGBLIGHT_EFFECT_RAINBOW_MOOD
#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
#define RGBLIGHT_EFFECT_SNAKE
#define RGBLIGHT_EFFECT_KNIGHT
#define RGBLIGHT_EFFECT_CHRISTMAS
#define RGBLIGHT_EFFECT_STATIC_GRADIENT
#define RGBLIGHT_EFFECT_RGB_TEST
#define RGBLIGHT_EFFECT_ALTERNATING
#define RGBLIGHT_EFFECT_TWINKLE
```

File: keyboards/handwired/nozbe_macro/info.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
+{
+    "keyboard_name": "nozbe_macro",
+    "url": "",
+    "maintainer": "Leon Omelan",
+    "width": 4,
+    "height": 1,
```
Comment (drashna) [inline keyboards/handwired/nozbe_macro/info.json:6]:
```suggestion
```

File: keyboards/handwired/nozbe_macro/rules.mk [hunk 3]
```diff
@@ -0,0 +1,19 @@
+# MCU name
+MCU = atmega32u4
+
+# Boot Section Size in *bytes*
+OPT_DEFS += -DBOOTLOADER_SIZE=4096
```
Comment (drashna) [inline keyboards/handwired/nozbe_macro/rules.mk:5]:
```suggestion
# Bootloader Section
BOOTLOADER = atmel-dfu
```

File: keyboards/handwired/nozbe_macro/rules.mk [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+# Build Options
+#   comment out to disable the options.
+BOOTMAGIC_ENABLE = yes	# Virtual DIP switch configuration
```
Comment (drashna) [inline keyboards/handwired/nozbe_macro/rules.mk:9]: There is no "full" bootmagic anymore, only the "lite" option, which can be turned on with just "yes".
```suggestion
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
```

File: keyboards/handwired/nozbe_macro/rules.mk [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+MOUSEKEY_ENABLE = yes	# Mouse keys
+EXTRAKEY_ENABLE = yes	# Audio control and System control
+CONSOLE_ENABLE = no	# Console for debug
+COMMAND_ENABLE = no    # Commands for debug and configuration
+SLEEP_LED_ENABLE = no  # Breathing sleep LED during USB suspend
+NKRO_ENABLE = yes		# USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
```
Comment (drashna) [inline keyboards/handwired/nozbe_macro/rules.mk:15]:
```suggestion
NKRO_ENABLE = yes       # USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
```

File: keyboards/handwired/nozbe_macro/rules.mk [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+BACKLIGHT_ENABLE = yes  # Enable keyboard backlight functionality
+AUDIO_ENABLE = no
+RGBLIGHT_ENABLE = no
+ENCODER_ENABLE = no
```
Comment (drashna) [inline keyboards/handwired/nozbe_macro/rules.mk:19]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by rozPierog** [comment]
Thanks for the review @drashna I've fixed the issues. Sorry about having so much old stuff in the code, I must have based my keyboard of some very old template. Let me know if there is anything else that I need to look at

File: keyboards/handwired/nozbe_macro/keymaps/default/keymap.c [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
+
+enum layer_names { _BASE, _FN };
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [_BASE] = LAYOUT(LT(_FN, KC_N), LALT(KC_UP), LALT(KC_DOWN), LALT(KC_D)),
+    [_FN]   = LAYOUT(KC_TRNS, KC_1, KC_2, BL_STEP),
```
Comment (drashna) [inline keyboards/handwired/nozbe_macro/keymaps/default/keymap.c:23]: This is more of a nitpick for readability, but: 
```suggestion
enum layer_names { 
    _BASE, 
    _FN,
};

const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
    [_BASE] = LAYOUT(
        LT(_FN, KC_N), LALT(KC_UP), LALT(KC_DOWN), LALT(KC_D)
    ),
    [_FN]   = LAYOUT(
        KC_TRNS, KC_1, KC_2, BL_STEP
    ),
```

**Review by drashna — APPROVED** [review]

File: keyboards/handwired/nozbe_macro/readme.md [hunk 8]
```diff
@@ -0,0 +1,26 @@
+# Nozbe Macropad handwired
+
+![Nozbe Macropad](https://i.imgur.com/fUMtWma.png)
+
+*4 key macropad with optional rotary encoder*
+
+* Keyboard Maintainer: [rozPierog](https://github.com/rozPierog)
+* Hardware Supported: *Custom handwired 4 key (or 3 key + rotary encoder) pad*
```
Comment (fauxpark) [inline keyboards/handwired/nozbe_macro/readme.md:8]:
```suggestion
![Nozbe Macropad](https://i.imgur.com/fUMtWmah.png)

4 key macropad with optional rotary encoder.

* Keyboard Maintainer: [rozPierog](https://github.com/rozPierog)
* Hardware Supported: Custom handwired 4 key (or 3 key + rotary encoder) pad
```

File: keyboards/handwired/nozbe_macro/rules.mk [hunk 9]
```diff
@@ -0,0 +1,18 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader Section
```
Comment (fauxpark) [inline keyboards/handwired/nozbe_macro/rules.mk:4]:
```suggestion
# Bootloader selection
```

File: keyboards/handwired/nozbe_macro/rules.mk [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+MOUSEKEY_ENABLE = yes   # Mouse keys
+EXTRAKEY_ENABLE = yes   # Audio control and System control
+CONSOLE_ENABLE = no     # Console for debug
+COMMAND_ENABLE = no     # Commands for debug and configuration
+SLEEP_LED_ENABLE = no   # Breathing sleep LED during USB suspend
+NKRO_ENABLE = yes       # USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+BACKLIGHT_ENABLE = yes  # Enable keyboard backlight functionality
+AUDIO_ENABLE = no
+RGBLIGHT_ENABLE = no
+
```
Comment (fauxpark) [inline keyboards/handwired/nozbe_macro/rules.mk:18]:
```suggestion
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = yes           # USB Nkey Rollover
BACKLIGHT_ENABLE = yes      # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
AUDIO_ENABLE = no           # Audio output
```

File: keyboards/handwired/nozbe_macro/nozbe_macro.h [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+#pragma once
+
+#include "quantum.h"
+
+#define LAYOUT(k00, k01, k02, k03) \
+    {                              \
+        { k00, k01, k02, k03 }    \
+    }
+
+#endif
```
Comment (fauxpark) [inline keyboards/handwired/nozbe_macro/nozbe_macro.h:26]:
```suggestion
#define LAYOUT( \
    k00, k01, k02, k03 \
) { \
    { k00, k01, k02, k03 } \
}
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by rozPierog** [comment]
Thanks @fauxpark for the review. I've committed changes you've suggested

**Comment by rozPierog** [comment]
Hey @fauxpark could you look at this again? I've fixed the issues you've reported in Code Review

**Review by fauxpark — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**